fastval.go

```diff
@@ -538,3 +544,29 @@ func NewRegexpFastVal(value *regexp.Regexp) FastVal {
 	}
 	return val
 }
+
+func NewPcreFastVal(value *pcre.Regexp) FastVal {
+	pcreMatcher := &PcreMatcherWrapper{
+		matcher:   &pcre.Matcher{},
+		pcreRegex: value,
+	}
+	val := FastVal{
+		dataType: PcreValue,
+		data:     pcreMatcher,
+	}
+	return val
+}
+
+type PcreMatcherWrapper struct {
+	matcher   *pcre.Matcher
+	pcreRegex *pcre.Regexp
+}
+
+func (wrapper *PcreMatcherWrapper) Match(data []byte) bool {
+	wrapper.matcher.Reset(*wrapper.pcreRegex, data, 0)
```

Comment:
It's probably worthwhile to create a new matcher object for each match (unless the cost is prohibitive).  By using a stored object like this, the same expression won't allow you to clone the match tree across multiple independant threads without triggering data use conflicts.

Reply:
Good point. I'll make the change.